test/src/common/bridgePowerController_ut.cpp

```diff
@@ -147,6 +147,287 @@ TEST_F(BridgePowerControllerTest, alignment) {
   EXPECT_EQ(next, 5100);
 }
 
+// Sample duration ends at 19 minutes, which is after subsample duration ends at 16 minutes,
+// and before the next aligned sample interval starts at 20 minutes.
+TEST_F(BridgePowerControllerTest, subsampling1) {
+  const uint32_t kTwentyMinutes = 1200000;
+  const uint32_t kNineteenMinutes = 1140000;
+  const uint32_t kFiveMinutes = 300000;
+  const uint32_t kOneMinute = 60000;
+  BridgePowerController powerController(FAKE_VBUS_EN, kTwentyMinutes, kNineteenMinutes,
+                                        kFiveMinutes, kOneMinute, true, true);
```

Comment:
This test case is exactly the configuration that exposed the bug in our RBR testing.